### PR TITLE
feat(HMS-2304): Nest list endpoints with 'data'

### PR DESCRIPTION
**Pull request by adiabramovitch**
This marks the initial step towards implementing pagination.
- I've introduced a "data" nesting structure for all list endpoints. 
-  I've eliminated the unnecessary "Payload" from the generic reservation variable. 

Please note that this is a significant change. I have thoroughly tested each endpoint, and everything appears to be functioning correctly. However,  PLEASE review the changes carefully as there's a possibility that I may have missed something.

Needs:
 - [x] IQE ready
 - [x] UI ready
 - [x] IB ready (https://github.com/RedHatInsights/image-builder-frontend/pull/1299)

### CODE REVIEW

**Review by ezr-ondrej — COMMENTED** [review]
I'd rather name the variables `XListResponse` instead of `XDataResponse` to follow the naming principle of naming after type purpose not it's inner structure.

But that is mostly a nit. Apart of it looks good to me :+1:

**Review by ezr-ondrej — APPROVED** [review]
I'm good here (we shall not merge until everyting else is ready tho)

**Comment by lzap** [comment]
OpenAPI for the last commit: https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/RHEnVision/provisioning-backend/0ad553e8ca6565d4b21c28a99c864cdfd352860c/api/openapi.gen.json#tag/Source/operation/getSourceAccountIdentity

**Review by lzap — APPROVED** [review]
Perfect, thanks.

**Comment by adiabramovitch** [comment]
> Hmmm... I am getting the following locally with the new spec.
> 
> ```
> Invalid type for variable 'received_data'. Required value type is list and passed type was dict at ['received_data']
> ```

The type we return according to spec is not an array anymore but an 'object'. 
so it is a dict and not a list.
A dict with one key: "data" which holds an array.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]

File: cmd/spec/path.yaml [hunk 1]
```diff
@@ -116,9 +116,9 @@ paths:
           content:
             application/json:
               schema:
-                type: array
+                type: object
                 items:
```
[the remark this answers is not in the document]
Comment (adiabramovitch) [reply cmd/spec/path.yaml]: fixed, @akhil-jha Maybe that will solve your problem?

File: cmd/spec/path.yaml [hunk 2]
```diff
@@ -148,9 +146,9 @@ paths:
           content:
             application/json:
               schema:
-                type: array
+                type: 
                 items:
```
Comment (ezr-ondrej) [inline cmd/spec/path.yaml:150]: I guess we forgot it here @adiabramovitch ? :)

**Review by ezr-ondrej — COMMENTED** [review]
I believe we missed one :stuck_out_tongue:

File: cmd/spec/path.yaml (hunk 2 above)
Comment (adiabramovitch) [reply cmd/spec/path.yaml]: Sorry! fixed it

**Comment by lzap** [comment]
[FInal version of the OpenAPI spec](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/RHEnVision/provisioning-backend/512b3a7bb643eaf87e58fad2428726547b853c88/api/openapi.gen.json).

**Comment by lzap** [comment]
IB patch merged.

**Review by lzap — APPROVED** [review]
Merging.

**Pull request merged**